### PR TITLE
CAMS-219 Case Docket API

### PR DESCRIPTION
**Pull request by btposey**
# Purpose

Provides case docket from DXTR to the user interface.

# Major Changes

* Implements the case-docket endpoint without search capability.
* Moved App Insight initialization to a common function.
* Added CAMS NotFoundError.
* Added decomposeCaseId function.
* Added createMockApplicationContext helper function to testing utilities module.
* Added logger helper API to the application context.
* Modified httpError response to include success == false.

# Testing/Validation

* Verified the API returned dockets (normal)and 404 not found (error) responses as expected.
* 100% line and branch coverage for case-docket specific source code added.
* Increased codebase coverage for line 93% and branch 84%.

### CODE REVIEW

File: backend/functions/case-docket/case-docket.function.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { httpError, httpSuccess } from '../lib/adapters/utils/http-response';
+import { applicationContextCreator } from '../lib/adapters/utils/application-context-creator';
+import { CaseDocketController } from '../lib/controllers/case-docket/case-docket.controller';
+import { initializeApplicationInsights } from '../azure/app-insights';
+
+dotenv.config();
+
+initializeApplicationInsights();
+
+//const MODULE_NAME = 'CASE-DOCKET-FUNCTION';
```
Comment (btposey) [inline backend/functions/case-docket/case-docket.function.ts:13]: Remove this if it isn't used.

File: backend/functions/lib/adapters/gateways/dxtr/case-docket.dxtr.gateway.test.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        results: {
+          recordset: DOCKET,
+        },
+        message: '',
+      };
+      querySpy.mockImplementation(async () => {
+        return Promise.resolve(mockResults);
+      });
+      const gateway: DxtrCaseDocketGateway = new DxtrCaseDocketGateway();
+      // TODO: Use new create mock context function.
```
Comment (btposey) [inline backend/functions/lib/adapters/gateways/dxtr/case-docket.dxtr.gateway.test.ts:50]: Should we just deal. with these TODOs now?

File: backend/functions/lib/use-cases/case-docket/case-docket.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { CaseDocketGateway } from '../../adapters/gateways/gateways.types';
+import { ApplicationContext } from '../../adapters/types/basic';
+import { CaseDocket } from './case-docket.model';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const MODULE_NAME = 'CASE-DOCKET-USE-CASE';
```
Comment (btposey) [inline backend/functions/lib/use-cases/case-docket/case-docket.ts:6]: Feels like we should keep this around for consistency, but it ISN'T used. Delete?

File: backend/functions/lib/adapters/types/basic.d.ts [hunk 4]
```diff
@@ -9,9 +10,19 @@ export interface AppConfig {
...
 
+export interface LoggerHelper {
+  debug: (moduleName: string, message: string, data?: unknown) => void;
+  info: (moduleName: string, message: string, data?: unknown) => void;
+  warn: (moduleName: string, message: string, data?: unknown) => void;
+  error: (moduleName: string, message: string, data?: unknown) => void;
+  camsError(erro: CamsError);
+}
+
+// TODO: By extending Context from Azure Functions we have allowed Azure to invade this codebase.
```
Comment (jamesobrooks) [inline backend/functions/lib/adapters/types/basic.d.ts:21]: We've wrapped the Azure `Context` object. What other way would we want to limit our exposure to risk from this third-party component?

I suppose we could write an interface and an adapter, but that adapter is still going to either wrap the Azure `Context` or we'll have to destructure it.

File: backend/functions/lib/common-errors/common-errors.test.ts [hunk 5]
```diff
@@ -23,4 +24,10 @@ describe('Common errors', () => {
     expect(error.module).toBe(testModuleName);
     expect(error.message).toBe('Unknown error');
   });
+  test('NotFoundError contructor', async () => {
```
Comment (jamesobrooks) [inline backend/functions/lib/common-errors/common-errors.test.ts:27]:
```suggestion

  test('NotFoundError contructor', async () => {
```

Comment (jamesobrooks) [reply backend/functions/lib/common-errors/common-errors.test.ts:27]: Could we add a newline here?

File: backend/functions/lib/adapters/gateways/dxtr/case-docket.dxtr.gateway.test.ts [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+      const mockContext: ApplicationContext = await applicationContextCreator(
+        {} as unknown as Context,
+      );
```
Comment (fmaddenflx) [inline backend/functions/lib/adapters/gateways/dxtr/case-docket.dxtr.gateway.test.ts:73]: It looks like you already are doing this.  Are you not?

File: backend/functions/jest.config.js [hunk 7]
```diff
@@ -7,17 +7,19 @@ module.exports = {
   coverageThreshold: {
     global: {
       lines: 90,
-      branches: 80,
+      branches: 85,
```
Comment (jamesobrooks) [inline backend/functions/jest.config.js:10]: Woot woot!

**Review by fmaddenflx — APPROVED** [review]

**Review by jamesobrooks — APPROVED** [review]

**Pull request merged**